_posts/2023-02-19-a-lean-dataops-platform.md

```diff
@@ -0,0 +1,158 @@
+
+### Dagster Orchestration 
+Similar to the Fivetran Terraform provider, we also wanted to trial the use of Dagster, another somewhat new and green-field technology based around fixing the shortcomings of tools like Airflow. Dagster has various integrations with the aforementioned technology we are already using. In addition, the use of dagster meant we could ensure, in code, that our DBT project was dependent on the success of our Fivetran pipeline data syncs. 
+
+Dagster for our purposes was very simple to set up, minimal lines of code are required to create dependencies between Fivetran and DBT. Dagster, which is run on a Github actions CRON job, first initiates a sync for all Fivetran connectors pointed at our destination and polls them until completion. Once successful the fivetran syncs are complete the DBT project is run, and then finally the tests for the DBT project are executed. Here is a diagram of Dagster and its interactions with Fivetran and DBT.
+
+![Dagster diagram](../img/data-driven-platform-images/dagster_diagram.png)
+
+
+# Data-Ops
+## CI/CD Pipeline
+
+In addition to our infrastructure being defined as code, another form that data-ops practices take in this platform is our continuous integration and continuous delivery pipeline. Running on Github actions we have two main workflows. The first is for all pull requests to the main branch. This deployment doesn't make any physical changes to our infrastructure, but instead does all the checks and balances to ensure nothing goes wrong when a branch is merged into main (being production). These checks include those discussed above for the Snowflake governance piece of our platform. In addition, they also run Terraform plans, which if successful indicate the proposed changes to be made will be applied successfully. The second workflow is the deployment of changes on the main branch. The steps in this deployment make changes to each Terraform workspace for Snowflake and Fivetran. With this structure we are able to make small incremental changes, which in rare cases they fail, can be rolled back quickly and easily.
+
+## Material outcomes, Ad-hoc Analysis and Dashboard 
+This was all setup in a couple of months, which could have been significantly longer if we chose some different technologies, in my opinion. The facilitation of ad-hoc is possible however, is yet to be utilized. We do plan to complete a customer segmentation piece of analysis in the coming weeks, so stay tuned for that blog. However, we have a complete and functional dashboard that illustrates our high level metrics.
```

Comment:
The facilitation of ad-hoc is possible however, is yet to be utilized

no idea what this means

Reply:
same